lib/plugins/rotate_asg_instances/asg.rb

```diff
@@ -48,6 +63,22 @@ def outdated_instances
 
     private
 
+    def ssh_executed_successfully?
+      Retriable.retriable(base_interval: 5, tries: 3) do
+        @command_response = run_ssh_command(first_ec2_instance, '/bin/true')
+        @command_response.exitstatus.zero?
+      end
+    end
+
+    def run_ssh_command(instance, command)
+      remote = "#{@ssh_user}@#{instance.public_dns_name}"
+      options = ['UserKnownHostsFile=/dev/null', 'StrictHostKeyChecking=no']
+      remote_cmd = "ssh -o #{options.join(' -o ')} #{remote} " \
+                   "-o ConnectTimeout=10 #{command}"
```

Comment:
We don't want to always add ConnectTimeout option, only for doctor and pre update checks, where we test the connection. When we run shutdown, let's skip it.